chain/rosetta-rpc/src/adapters/mod.rs

```diff
@@ -144,79 +146,313 @@ pub(crate) async fn convert_block_to_transactions(
+    /// mapping from transaction hash to signed id.  This first call is quite
+    /// expensive.  Subsequent calls are just lookups in a map.
+    async fn get_signer_for(
+        &mut self,
+        tx_hash: &near_primitives::hash::CryptoHash,
+    ) -> crate::errors::Result<near_primitives::types::AccountId> {
+        let signers = if let Some(ref signers) = self.signers {
+            signers
+        } else {
+            let mut map = std::collections::HashMap::new();
+            for chunk in self.block.chunks.iter() {
+                let chunk_hash = near_primitives::sharding::ChunkHash(chunk.chunk_hash);
+                let transactions = self
+                    .view_client_addr
+                    .send(near_client::GetChunk::ChunkHash(chunk_hash))
+                    .await?
+                    .map_err(|err| {
+                        crate::errors::ErrorKind::InternalError(format!(
+                            "Failed to fetch chunk: {}",
+                            err
+                        ))
+                    })?
+                    .transactions;
+                for tx in transactions {
+                    map.insert(tx.hash, tx.signer_id);
+                }
+            }
+            self.signers.insert(map)
+        };
+        signers
+            .get(tx_hash)
+            .ok_or_else(|| {
+                crate::errors::ErrorKind::InternalError(format!(
+                    "Unable to find signer for transaction: {}",
+                    tx_hash
+                ))
+            })
+            .map(Clone::clone)
+    }
+
+    /// Returns outcome of a transaction.  Waits if it’s not recorded on the
+    /// chain.
+    async fn get_transaction_outcome(
+        &mut self,
+        tx_hash: near_primitives::hash::CryptoHash,
+        signer: near_primitives::types::AccountId,
+    ) -> crate::errors::Result<near_primitives::views::FinalExecutionOutcomeWithReceiptView> {
+        use near_primitives::views::FinalExecutionOutcomeViewEnum::*;
+
+        tokio::time::timeout(self.polling_config.polling_timeout, async {
+            loop {
+                let status = self
+                    .view_client_addr
+                    .send(near_client::TxStatus {
+                        tx_hash: tx_hash.clone(),
+                        signer_account_id: signer.clone(),
+                        fetch_receipt: true,
+                    })
+                    .await?
+                    .map_err(|err| {
+                        crate::errors::ErrorKind::InternalError(format!(
+                            "Failed to fetch transaction: {}: {:?}",
+                            tx_hash, err
+                        ))
+                    })?;
+                if let Some(outcome) = status {
+                    break Ok(outcome);
+                }
+                tokio::time::sleep(self.polling_config.polling_interval).await;
+            }
+        })
```

Comment:
We don't need the whole TxStatus (which pulls and resolves all the receipts recursively and thus will block Rosetta block response until all the future receipts get executed).

Given that all you want to know is the list of produced receipts, you should be able to re-use (or consider moving it into ViewClient) what we have in Indexer Framework to deal with the [local receipts, and delayed local receipts](https://github.com/near/nearcore/blob/1037931fc45b3f51fb512bc9cb61ac663c8eb0c5/chain/indexer/src/streamer/mod.rs#L67-L161)